### PR TITLE
Move SITL on hardware build scripts to Tools scripts directory

### PR DESCRIPTION
**Pull request by Hwurzburg**
Moves the script and files to Tools....
Revises the hwdef and default.parm to what is really required, this also greatly increases the space on 1MB boards running this since GPS and RFs are eliminated....Pixhawk1-1M has 32KB free running SITL on HW
also needs Peters #23779 along with this change to the AP_AHRS_Backend.h in order to compile on 1MB boards...

### CODE REVIEW

File: libraries/AP_AHRS/AP_AHRS_Backend.h [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #include <inttypes.h>
 #include <AP_Airspeed/AP_Airspeed.h>
 #include <AP_InertialSensor/AP_InertialSensor.h>
+#include <AP_NavEKF/AP_Nav_Common.h>
```
Comment (peterbarker) [inline libraries/AP_AHRS/AP_AHRS_Backend.h:27]: Commit message is not correct for this change

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_AHRS/AP_AHRS_Backend.h (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_AHRS/AP_AHRS_Backend.h:27]: fixed

[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_AHRS/AP_AHRS_Backend.h:27]: that fails to compile....
[  11/1062] Compiling libraries/AC_AttitudeControl/ControlMonitor.cpp
[  13/1062] Compiling libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp
[  15/1062] Compiling libraries/AC_AttitudeControl/AC_AttitudeControl_TS.cpp
[  16/1062] Compiling libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp
In file included from ../../libraries/AP_AHRS/AP_AHRS_DCM.h:24,
                 from ../../libraries/AP_AHRS/AP_AHRS.h:32,
                 from ../../libraries/AP_AHRS/AP_AHRS_View.h:23,
                 from ../../libraries/AC_AttitudeControl/AC_AttitudeControl.h:9,
                 from ../../libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.h:6,
                 from ../../libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp:1:
../../libraries/AP_AHRS/AP_AHRS_Backend.h:287:43: error: 'struct' tag used in naming 'union nav_filter_status' [-fpermissive]
  287 |     virtual bool get_filter_status(struct nav_filter_status &status) const {
      |                                           ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.

[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_AHRS/AP_AHRS_Backend.h]: fixed using union instead

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**